spec.bs

```diff
@@ -2788,6 +2824,159 @@ The <dfn for=Navigator method>canLoadAdAuctionFencedFrame()</dfn> method steps a
+  : <dfn>ad render ID</dfn>
+  :: A [=string=] containing the [=interest group ad/ad render ID=] for the ad represented by this entry.
+</dl>
+
+A <dfn>server auction request context</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="server auction request context">
+  : <dfn>request ID</dfn>
+  :: A unique identifier associated with this and only this invocation of
+    {{Window/navigator}}.{{Navigator/getInterestGroupAdAuctionData()}}. This is
+    used to look-up a specific request context.
+  : <dfn>request context</dfn>
+  :: An opaque context used to handle the request.
+</dl>
+
+<div algorithm="getInterestGroupAdAuctionData()">
+
+The <dfn for=Navigator method>getInterestGroupAdAuctionData(|config|)</dfn> method steps are:
+
+1. Let |global| be [=this=]'s [=relevant global object=].
+1. If |global|'s [=associated Document=] is not [=allowed to use=] the "[=run-ad-auction=]"
+  [=policy-controlled feature=], then [=exception/throw=] a "{{NotAllowedError}}" {{DOMException}}.
+1. Let |p| be [=a new promise=].
+1. Let |queue| be the result of [=starting a new parallel queue=].
+1. [=parallel queue/enqueue steps|Enqueue the following steps=] to |queue|:
+  1. Let |igMap| be a new [=map=] whose [=map/keys=] are [=origins=] and [=map/values=] are [=lists=].
+  1. Let |startTime| be a [=moment=] equal to the [=current wall time=].
+  1. [=list/For each=] |ig| of the [=user agent=]'s [=interest group set=]:
+    1. If |ig|'s [=interest group/ads] is null or [=list/is empty=], [=iteration/continue=].
+    1. Let |owner| be |ig|'s [=interest group/owner=].
+    1. If |igMap|[|owner|] does not [=map/exist=]:
+      1. [=map/Set=] |igMap|[|owner|] to a new [=list=].
+    1. Let |ads| be a new [=list=].
+    1. [=list/For each=] |ad| in |ig|'s [=interest group/ads=], [=list/Append=] |ad|'s [=interest group ad/ad render ID=] to |ads|.
+    1. Let |components| be a new [=list=].
+    1. [=list/For each=] |component| in |ig|'s [=interest group/ad components=], [=list/append=] |component|'s [=interest group ad/ad render ID=] to |components|.
+    1. Let |prevWins| be a new <code>[=sequence=]<[=server auction previous win=]></code>.
+    1. [=list/For each=] |prevWin| of |ig|'s [=interest group/previous wins=] for all days within the
+      the last 30 days:
+      1. Let |timeDelta| be |startTime| minus |prevWin|'s [=previous win/time=].
+      1. Set |timeDelta| to 0 if |timeDelta| is negative, |timeDelta|'s nearest second (rounding down)
+        otherwise.
+      1. Let |serverPrevWin| be a new [=server auction previous win=] with the following [=struct/items=]:
+        : [=server auction previous win/time delta=]
+        :: |timeDelta|
+        : [=server auction previous win/ad render ID=]
+        :: the value of the "adRenderId" field in |prevWin|'s [=previous win/ad json=], or the empty string if not present

```

Comment:
@domfarolino Two questions.
1. the prevWin's ad json has an adRenderId field in [implementation](https://source.chromium.org/chromium/chromium/src/+/main:content/services/auction_worklet/public/mojom/bidder_worklet.mojom;drc=060a904b3f09958d2c424d7cbbfe729bf6747e8e;l=32)). But the [spec](https://github.com/WICG/turtledove/blob/main/spec.bs#L1423) used [serializing an Infra value to a JSON string](https://infra.spec.whatwg.org/#serialize-an-infra-value-to-a-json-string) to convert 
an [interest group ad](https://wicg.github.io/turtledove/#interest-group-ad) struct (a subset of its fields)  to a JSON. Does it directly use the struct's field name as the key in json? If so, we should change these field names to match the implementation, i.e., renderURL, adRenderId, etc,.?
2. Is it OK to write something like "the value of the "adRenderId" field in a json"?